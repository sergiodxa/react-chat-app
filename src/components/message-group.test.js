import React from "react";
import { render } from "@testing-library/react";
import MessageGroup from "./message-group";

test("MessageGroup", () => {
  const scrollIntoViewMock = jest.fn();
  window.HTMLElement.prototype.scrollIntoView = scrollIntoViewMock;

  const messages = [
    {
      content: "hola",
      username: "sergiodxa",
      date: "2019-06-02T00:45:43.963Z",
      channel: "general"
    },
    {
      content: "hola, que tal?",
      username: "sergiodxa",
      date: "2019-06-01T00:45:43.963Z",
      channel: "general"
    },
    {
      content: "hola, que tal?",
      username: "sergiodxa",
      date: "2019-05-03T00:45:43.963Z",
      channel: "general"
    }
  ];

  const { asFragment } = render(
    <MessageGroup now={new Date()} messages={messages} />
  );

  expect(asFragment()).toMatchSnapshot();

  expect(scrollIntoViewMock).toHaveBeenCalled();
});
