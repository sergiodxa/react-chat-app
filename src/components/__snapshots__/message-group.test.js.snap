// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageGroup 1`] = `
<DocumentFragment>
  .emotion-2 {
  position: relative;
  margin: 2rem 0;
}

.emotion-0 {
  background: white;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  margin: 0 auto;
  font-size: 1rem;
  font-weight: normal;
  text-align: center;
  width: 20%;
}

.emotion-1 {
  border: 0;
  border-top: 1px solid #ccc;
}

.emotion-6 {
  padding: 1rem 1.5rem;
  border-bottom: 1px solid #eaeaea;
}

.emotion-6:last-child {
  border-bottom: none;
}

.emotion-3 {
  font-size: 1.1rem;
}

.emotion-4 {
  font-size: .75rem;
  font-style: normal;
  color: #666;
}

.emotion-5 {
  margin: .5rem 0 0;
}

<section>
    <header
      class="emotion-2"
    >
      <h3
        class="emotion-0"
      >
        Saturday, June 1st
      </h3>
      <hr
        class="emotion-1"
      />
    </header>
    <section>
      <article
        class="emotion-6"
      >
        <div>
          <strong
            class="emotion-3"
          >
            sergiodxa
          </strong>
           
          <time
            class="emotion-4"
          >
            19:45:43
          </time>
        </div>
        <p
          class="emotion-5"
        >
          hola
        </p>
      </article>
    </section>
    <header
      class="emotion-2"
    >
      <h3
        class="emotion-0"
      >
        Friday, May 31st
      </h3>
      <hr
        class="emotion-1"
      />
    </header>
    <section>
      <article
        class="emotion-6"
      >
        <div>
          <strong
            class="emotion-3"
          >
            sergiodxa
          </strong>
           
          <time
            class="emotion-4"
          >
            19:45:43
          </time>
        </div>
        <p
          class="emotion-5"
        >
          hola, que tal?
        </p>
      </article>
    </section>
    <header
      class="emotion-2"
    >
      <h3
        class="emotion-0"
      >
        Thursday, May 2nd
      </h3>
      <hr
        class="emotion-1"
      />
    </header>
    <section>
      <article
        class="emotion-6"
      >
        <div>
          <strong
            class="emotion-3"
          >
            sergiodxa
          </strong>
           
          <time
            class="emotion-4"
          >
            19:45:43
          </time>
        </div>
        <p
          class="emotion-5"
        >
          hola, que tal?
        </p>
      </article>
    </section>
  </section>
</DocumentFragment>
`;
